nil/services/txnpool/txnpool.go

```diff
@@ -16,30 +16,34 @@ import (
 
 type Pool interface {
 	Add(ctx context.Context, txns ...*types.Transaction) ([]DiscardReason, error)
-	Discard(ctx context.Context, txns []*types.Transaction, reason DiscardReason) error
+	Discard(ctx context.Context, txns []common.Hash, reason DiscardReason) error
 	OnCommitted(ctx context.Context, committed []*types.Transaction) error
 	// IdHashKnown check whether transaction with given Id hash is known to the pool
 	IdHashKnown(hash common.Hash) (bool, error)
 	Started() bool
 
-	Peek(ctx context.Context, n int) ([]*types.Transaction, error)
+	Peek(ctx context.Context, n int) ([]*MetaTxn, error)
 	SeqnoToAddress(addr types.Address) (seqno types.Seqno, inPool bool)
 	TransactionCount() int
 	Get(hash common.Hash) (*types.Transaction, error)
 }
 
-type metaTxn struct {
+type MetaTxn struct {
```

Comment:
Initially I considered this as a pure internal txpool type.
We probably can add more system fields in future that will be handled. Maybe let's add something like `TxWithHash` to `types` and inline it here?

I'd avoid leaking `MetaTxn` type outside the pool.